job-app/src/components/popup/Popup.js

```diff
@@ -0,0 +1,82 @@
+import './Popup.css';
+import React from 'react';
+
+import DatePicker from 'react-datepicker';
+import "react-datepicker/dist/react-datepicker.css";
+import {v4 as uuidv4} from 'uuid';
+
+class Popup extends React.Component {
+
+    constructor(props) {
+        super();
+        this.state = {
+            addCard: props.addCard,
+            cancel: props.cancel,
+            interviewDate: new Date()
+        }
+        this.addMyCard = this.addMyCard.bind(this);
+        this.cancel = this.cancel.bind(this);
+    }
+
+    addMyCard() {
+        const gcn = (target) => {
+            return document.getElementsByClassName(target)[0].value;
+        };
+
+        const assignedCategories = gcn("categories").split(",").map((cat) => {
+            const ret = cat.trim();
+            if (ret[0] !== '#') {
+                ret = "#" + ret;
+            }
+            return ret;
+        });
+
+        const card = {
+            ApplicationID: uuidv4(),
+            Position: gcn("position"),
+            Salary: gcn("salary"),
+            CompanyName: gcn("company-name"),
+            CompanyPhone: gcn("company-phone"),
+            CompanyAddress: gcn("company-address"),
+            Categories: assignedCategories,
+            AppliedDate: this.state.interviewDate.toISOString(),
+            Description: gcn("description")
+        };
+
+        this.state.addCard(card);
+    }
+
+    cancel() {
+        this.state.cancel();
+    }
+
+    render() {
+        return (
+            <div className="popup">
+                <div className="filler filler-horizontal">
+                    <div className="filler-buffer"></div>
+                    <div className="filler filler-vertical">
+                        <div className="filler-buffer"></div>
+                        <div className="popup-body">
+                            <div className="popup-description">Add an application.</div>
```

Comment:
do there need to be this many divs? It just looks weird that there's this many but all the inputs are within one

Reply:
The divs look like this because they are all building a scaffolding around the `popup-body`, essentially. If the css was working right it would look like this (imagine the fillers are transparent grey)
![image](https://user-images.githubusercontent.com/26638487/109876094-540f3a00-7c3f-11eb-94af-2f6a12f948eb.png)